src/frontend/src/handler/create_sql_function.rs

```diff
@@ -0,0 +1,180 @@
+                // return type is a struct for multiple columns
+                let datatypes = columns
+                    .iter()
+                    .map(|c| bind_data_type(&c.data_type))
+                    .collect::<Result<Vec<_>>>()?;
+                let names = columns
+                    .iter()
+                    .map(|c| c.name.real_value())
+                    .collect::<Vec<_>>();
+                return_type = DataType::new_struct(datatypes, names);
+            }
+            Kind::Table(TableFunction {})
+        }
+        None => {
+            return Err(ErrorCode::InvalidParameterValue(
+                "return type must be specified".to_string(),
+            )
+            .into())
+        }
+    };
+
+    let mut arg_types = vec![];
+    for arg in args.unwrap_or_default() {
+        arg_types.push(bind_data_type(&arg.data_type)?);
+    }
+
+    // resolve database and schema id
+    let session = &handler_args.session;
+    let db_name = session.database();
+    let (schema_name, function_name) = Binder::resolve_schema_qualified_name(db_name, name)?;
+    let (database_id, schema_id) = session.get_database_and_schema_id_for_create(schema_name)?;
+
+    // check if function exists
+    if (session.env().catalog_reader().read_guard())
+        .get_schema_by_id(&database_id, &schema_id)?
+        .get_function_by_name_args(&function_name, &arg_types)
+        .is_some()
+    {
+        let name = format!(
+            "{function_name}({})",
+            arg_types.iter().map(|t| t.to_string()).join(",")
+        );
+        return Err(CatalogError::Duplicated("function", name).into());
+    }
+
+    // check the service
+    // let client = ArrowFlightUdfClient::connect(&link)
+    //     .await
+    //     .map_err(|e| anyhow!(e))?;
+    // /// A helper function to create a unnamed field from data type.
+    // fn to_field(data_type: arrow_schema::DataType) -> arrow_schema::Field {
+    //     arrow_schema::Field::new("", data_type, true)
+    // }
+    // let args = arrow_schema::Schema::new(
+    //     arg_types
+    //         .iter()
+    //         .map::<Result<_>, _>(|t| Ok(to_field(t.try_into()?)))
+    //         .try_collect::<_, Fields, _>()?,
+    // );
+    // let returns = arrow_schema::Schema::new(match kind {
+    //     Kind::Scalar(_) => vec![to_field(return_type.clone().try_into()?)],
+    //     Kind::Table(_) => vec![
+    //         arrow_schema::Field::new("row_index", arrow_schema::DataType::Int32, true),
+    //         to_field(return_type.clone().try_into()?),
+    //     ],
+    //     _ => unreachable!(),
+    // });
+    // client
+    //     .check(&identifier, &args, &returns)
+    //     .await
+    //     .map_err(|e| anyhow!(e))?;
+
+    let function = Function {
+        id: FunctionId::placeholder().0,
+        schema_id,
+        database_id,
+        name: function_name,
+        kind: Some(kind),
+        arg_types: arg_types.into_iter().map(|t| t.into()).collect(),
```

Comment:
Looks like the type of this field should be changed to `Vec<Field>` or `Schema` so that we can find argument names here.

Reply:
Will change this field after reviewing the anonymous implementation logic